### PR TITLE
Adding enabled_protocol to azurerm_storage_share

### LINKED ISSUE #861: Add support for Azure File NFS support
TF provider version 2.84 and above added support for azure file enabled_protocol option located [here](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/storage_share#enabled_protocol) 

CAF is currently missing acl and provider options

### PR DESCRIPTION
**Pull request by hriaz**
Fixes #861 

Introduces support for Azure File share NFS support.  Requires minimum provider version 2.84

### CODE REVIEW

**Review by arnaudlh — CHANGES_REQUESTED** [review]
thanks for the contribution @hriaz, looks good to me, do you mind adding examples for CI/CD and also visibility? thanks!

**Comment by hriaz** [comment]
@arnaudlh Will do thanks.

**Comment by arnaudlh** [comment]
Testing and getting: 

╷
│ Error: Missing required argument
│ 
│   on ../modules/storage_account/file_share/file_share.tf line 15, in resource "azurerm_storage_share" "fs":
│   15:       dynamic "access_policy" {
│ 
│ The argument "for_each" is required, but no definition was found.
╵
Terraform plan return code: 1

@hriaz - Can you please have a look if you want this one merged? thanks!

**Comment by hriaz** [comment]
> Testing and getting:
> 
> ╷ │ Error: Missing required argument │ │ on ../modules/storage_account/file_share/file_share.tf line 15, in resource "azurerm_storage_share" "fs": │ 15: dynamic "access_policy" { │ │ The argument "for_each" is required, but no definition was found. ╵ Terraform plan return code: 1
> 
> @hriaz - Can you please have a look if you want this one merged? thanks!

Please test. Updated try block and added to CI including example

File: modules/storage_account/file_share/file_share.tf [hunk 1]
```diff
@@ -6,6 +6,23 @@ resource "azurerm_storage_share" "fs" {
   storage_account_name = var.storage_account_name
   quota                = try(var.settings.quota, null)
   metadata             = try(var.settings.metadata, null)
+  enabled_protocol     = try(var.settings.enabled_protocol, "SMB")
```
Comment (arnaudlh) [inline modules/storage_account/file_share/file_share.tf:9]:
```suggestion
  enabled_protocol     = try(var.settings.enabled_protocol, null)
```

File: examples/storage_accounts/108-file-share-nfs/configuration.tfvars [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    account_replication_type = "LRS"       # https://docs.microsoft.com/en-us/azure/storage/common/storage-redundancy
+    enable_https_traffic_only = false
+
+
+    tags = {
+      environment = "dev"
+      team        = "IT"
+    }
+
+    network = {
```
Comment (arnaudlh) [inline examples/storage_accounts/108-file-share-nfs/configuration.tfvars:30]: Comment that section or deployment does not succeed.

**Review by arnaudlh — CHANGES_REQUESTED** [review]
Almost there :)

**Review by arnaudlh — APPROVED** [review]
LGTM

**Pull request merged**